Logger/Logger.cc

```diff
@@ -0,0 +1,330 @@
+#include "Logger.h"
+
+#include <casacore/casa/OS/Directory.h>
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+void CreateLoggers(std::string log_filename) {
+    // Set a log file name and its path
+    if (!log_filename.empty()) {
+        try {
+            casacore::File tmp_filename(log_filename);
+            casacore::Directory tmp_dir = tmp_filename.path().dirName();
+            if (!tmp_dir.exists() || !tmp_dir.isWritable() || fs::path(log_filename).filename().empty() ||
+                !fs::is_regular_file(fs::status(log_filename))) {
+                log_filename = fs::path(getenv("HOME")).string() + "/CARTA/log/carta.log";
+                spdlog::warn("Can not create a log file! Use the default path name {0}", log_filename);
+            } else {
+                spdlog::info("Set the log file {0}", log_filename);
+            }
+        } catch (...) {
+            log_filename = fs::path(getenv("HOME")).string() + "/CARTA/log/carta.log";
+            spdlog::warn("Can not create a log file! Use the default path name {0}", log_filename);
+        }
+    } else {
+        log_filename = fs::path(getenv("HOME")).string() + "/CARTA/log/carta.log";
+        spdlog::info("Set the log file {0}", log_filename);
+    }
+
+    // Set a log file with its maximum size 5 MB and with 0 rotated file
+    auto log_file_sink = std::make_shared<spdlog::sinks::rotating_file_sink_mt>(log_filename, 1048576 * 5, 0);
+
+    // Set a log file pattern
+    log_file_sink->set_pattern("[%Y-%m-%d %H:%M:%S.%e] [%n] %v");
+
+    // Create a logger (tag RECEIVE) and save its messages to a log file
+    auto receive_logger = std::make_shared<spdlog::logger>("<<==", log_file_sink);
+    receive_logger->info("\">>>>>>>>> Start the RECEIVE logger <<<<<<<<<\"");
+
+    // Register the logger (tag RECEIVE) so we can get it globally
+    spdlog::register_logger(receive_logger);
+
+    // Create a logger (tag SEND) and save its messages to a log file
+    auto send_logger = std::make_shared<spdlog::logger>("==>>", log_file_sink);
+    send_logger->info("\">>>>>>>>> Start the SEND logger <<<<<<<<<\"");
+
+    // Register the logger (tag SEND) so we can get it globally
+    spdlog::register_logger(send_logger);
+}
+
+void LogReceiveEventType(uint16_t event_type) {
+    std::string logger_name = "<<==";
+    std::shared_ptr<spdlog::logger> receive_logger = spdlog::get(logger_name);
+    if (!receive_logger) {
+        spdlog::error("Fail to get the logger {0}", logger_name);
+        return;
+    }
+
+    switch (event_type) {
+        case CARTA::EventType::REGISTER_VIEWER: {
```

Comment:
please use an `unordered_map<CARTA::EventType, string>` instead of a very long switch